test/1.4/expressions/T_tqmic.dml

```diff
@@ -0,0 +1,90 @@
+/*
+  © 2024 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+
+// Test for Template-Qualified Method Implementation Calls (TQMIC)
+
+dml 1.4;
+device test;
+
+template t {
+    shared method sm() -> (double) default {
+        return 1.0;
+    }
+
+    shared method m() -> (double) default {
+        return 1.0;
+    }
+}
+
+template u1 is t {}
+
+template u2 is t {
+    shared method sm() -> (double) default {
+        return 2.2;
+    }
+
+    shared method m() -> (double) default {
+        return 2.2;
+    }
+}
+
+template u3 is t {
+    shared method sm() -> (double) default {
+        return 2.3;
+    }
+
+    param enable_u3 default true;
+    #if (enable_u3) {
+        method m() -> (double) default {
+            return 2.3;
+        }
+    }
+}
+
+// Redundancy as a smoke test that ancestor vtable merging works properly
+template v is (t, u1, u2, u3) {}
+
+template w is (t, u1, u2, u3, v) {
+    shared method sm() -> (double) default {
+        return 4.0;
+    }
+    method m() -> (double) {
+        return 4.0;
+    }
+}
+
+group g1 is w;
+group g2 is w {
+    param enable_u3 = false;
+}
+
+method init() {
```

Comment:
Perhaps also test e.g. `(&g.templates.w.sm)(dev.obj)`? Or is that banned for now?

Reply:
It's not, but why? There's nothing special about a TQMIC once it's resolved. At that point it's just a `NodeRef` to a method or a `TraitMethodDirect`, and you can do anything with it you could with a regular method reference.

Reply:
Everything that makes TQMIC special has to do with the resolution, and that's what the tests... test.